custom types migration

pbauer opened the pull request:
fix #133
custom migrations are still lack some nice-to-have features (export and import of field-configuration as json) but it is good enough for now.

pbauer commented:
Can someone please review this. It was a mayor effort and I'd really like to merge it soon. Many many thanks to @cekk @gbastien @tiazma and @flohcim! It was fun to work with you guys.

cekk commented:
@pbauer thank you too!
What's the status of this branch?
Last time that i worked on it, it was working more or less but i don't know if we can considerate this complete and mergeable

pbauer commented:
@cekk I fixed some bugs (we always only migrated one field when using the form) and added the feature to migrate Types without fti. That means we can now migrate ATEvents to DXEvents even though the old FTI was already replaced with the DX-version and can decide which fields we want to keep. That is also very useful for testing :-) 
The only mayor feature that is missing is the ability to export and import the migration-config as json.

pbauer commented:
I really wanted custom migrations to be shipped with Plone 5.0b2 so that everybody can try it.
